cmake/add_enclave_test.cmake

```diff
@@ -6,70 +6,9 @@
-        string(LENGTH ${OE_TESTS_PATH_NAME} OE_TESTS_PATH_NAME_LENGTH)
-        string(FIND ${CMAKE_CURRENT_SOURCE_DIR} ${OE_TESTS_PATH_NAME} TEST_DIR_INDEX)
-
-        if (${TEST_DIR_INDEX} GREATER -1)
-            math(EXPR TEST_DIR_INDEX "${TEST_DIR_INDEX}+${OE_TESTS_PATH_NAME_LENGTH}")
-            string(SUBSTRING ${CMAKE_CURRENT_SOURCE_DIR} ${TEST_DIR_INDEX} -1 TEST_DIR)
-        endif()
-
-        if (${TEST_DIR_INDEX} LESS_EQUAL -1 OR TEST_DIR STREQUAL "")
-            message(FATAL_ERROR "add_enclave_test can only be used in a subfolder of ${OE_TESTS_PATH_NAME} with ADD_WINDOWS_ENCLAVE_TESTS.")
-        endif()
-
-        # (HACK1) Ideally, the path to the enclave should be $<TARGET_FILE:${ENC_FILE}>
-        # However, for windows, the Linux build of the enclave is used for testing.
-        # Instead of passing in "enc" as the subpath, we are using "enc" as the default
-        # enclave subpath and "host" as the default host subpath.
-        # This hack can be removed when CMake on Windows produces ELF enclaves.
-        set(TEST_ENCSUBPATH enc)
-        set(TEST_HOSTSUBPATH host)
-
-        # (HACK2) This is a hack to figure out the target name for the linux enclave
-        # Ideally, the name of the enclave is found by $<TARGET_FILE:${ENC_FILE}>
-        # However, on windows, testing is done with the Linux build of the enclave.
-        # This hack can be removed when CMake on Windows produces ELF enclaves.
-        set(TEST_ENCFILE ${ENC_FILE})
-        if(ENC_FILE MATCHES ".*_signed")
-            string(REGEX REPLACE "_signed" ".signed" TEST_ENCFILE ${ENC_FILE})
-        endif()
-
-        # custom rule to copy binary from linux
-        # take a dependency on host binary to make sure it exists in addition to
-        # enc binary in linux
-        add_custom_command(OUTPUT ${TEST_NAME}_windows_include
-            COMMAND ${CMAKE_COMMAND} -E copy ${LINUX_BIN_DIR}/${TEST_DIR}/${TEST_ENCSUBPATH}/${TEST_ENCFILE} ${CMAKE_CURRENT_BINARY_DIR}/${TEST_HOSTSUBPATH}/${TEST_ENCFILE}
-            DEPENDS $<TARGET_FILE:${HOST_FILE}> ${LINUX_BIN_DIR}/${TEST_DIR}/${TEST_ENCSUBPATH}/${TEST_ENCFILE}
-            WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-            )
-
-        # add a custom target to ALL so that this step always needs to be run if
-        # this function is invoked
-        get_filename_component(TEST_NAME_WITHOUT_SLASH ${TEST_NAME} NAME)
-        add_custom_target(${TEST_NAME_WITHOUT_SLASH}.windows ALL
-            DEPENDS ${TEST_NAME}_windows_include
-            )
-
-        add_test(NAME ${TEST_NAME} COMMAND $<TARGET_FILE:${HOST_FILE}> ${CMAKE_CURRENT_BINARY_DIR}/${TEST_HOSTSUBPATH}/${TEST_ENCFILE} ${ARGN})
-
-    elseif (UNIX OR USE_CLANGW)
-        add_test(NAME ${TEST_NAME} COMMAND $<TARGET_FILE:${HOST_FILE}> $<TARGET_FILE:${ENC_FILE}> ${ARGN})
-    endif()
+    add_test(NAME ${TEST_NAME} COMMAND $<TARGET_FILE:${HOST_FILE}> $<TARGET_FILE:${ENC_FILE}> ${ARGN})
```

Comment:
Reading through this, I realized that @anitagov was correct and this would break the manual scenario she wanted to retain where the ability to build purely windows bits and then copy over linux enclave bits and run the tests.

My original thinking was that the `USE_CLANGW` flag would suffice, but that means that the only way the ctests get added is if you also did the cross compile build.

Instead of reverting this though, I suggest that we add a new flag to only build the windows components:
- It would be mutually exclusive to USE_CLANGW
- It would skip test enclaves and anything else that relied on bash, such as the test/crypto/data)
- It would require doing a pass through the CMakeLists.txt in the repo to respect this new flag.

Reply:
Thanks @CodeMonkeyLeet . This feature is very handy to enable ELF enclaves on Windows, especially the 3rd party libraries when issues are encountered using the cross-compiler.
Like the new flag suggested by @CodeMonkeyLeet